CBLClient/Apps/CBLTestServer-Dotnet/TestServer/PredictiveQueriesMethods.cs

```diff
@@ -0,0 +1,208 @@
+    {
+        #region Public Methods
+        public static void RegisterModel([NotNull] NameValueCollection args,
+                                  [NotNull] IReadOnlyDictionary<string, object> postBody,
+                                  [NotNull] HttpListenerResponse response)
+        {
+            var modelName = postBody["model_name"].ToString();
+            var echoModel = new EchoModel(modelName);
+            Database.Prediction.RegisterModel(modelName, echoModel);
+            response.WriteBody(MemoryMap.Store(echoModel));
+        }
+
+        public static void UnRegisterModel([NotNull] NameValueCollection args,
+                                  [NotNull] IReadOnlyDictionary<string, object> postBody,
+                                  [NotNull] HttpListenerResponse response)
+        {
+            var modelName = postBody["model_name"].ToString();
+            Database.Prediction.UnregisterModel(modelName);
+            response.WriteEmptyBody();
+        }
+
+        internal static void GetPredictionQueryResult([NotNull] NameValueCollection args,
+            [NotNull] IReadOnlyDictionary<string, object> postBody,
+            [NotNull] HttpListenerResponse response)
+        {
+            With<Database>(postBody, "database", db =>
+            {
+                With<EchoModel>(postBody, "model", model =>
+                {
+                    var dict = postBody["dictionary"];
+                    var input = Expression.Value(dict);
+                    var prediction = Function.Prediction(model.name, input);
+                    List<Object> resultArray = new List<Object>();
+                    using (IQuery query = QueryBuilder
+                           .Select(SelectResult.Expression(prediction))
+                           .From(DataSource.Database(db)))
+                        foreach (Result row in query.Execute())
+                        {
+                            resultArray.Add(row.ToDictionary());
+                        }
+                    response.WriteBody(resultArray);
+                });
+            });
+        }
+
+        internal static void NonDictionary([NotNull] NameValueCollection args,
+            [NotNull] IReadOnlyDictionary<string, object> postBody,
+            [NotNull] HttpListenerResponse response)
+        {
+            With<Database>(postBody, "database", db =>
+            {
+                With<EchoModel>(postBody, "model", model =>
+                {
+                    var dict = postBody["dictionary"];
+                    var input = Expression.Value(dict);
+                    var prediction = Function.Prediction(model.name, input);
+                    List<Object> resultArray = new List<Object>();
+                    using (IQuery query = QueryBuilder
+                           .Select(SelectResult.Expression(prediction))
+                           .From(DataSource.Database(db)))
+                        try
```

Comment:
is error handling needed here? what we will get is pass if execution fail with below handling. If we are expecting error as PASS, what will happen for no error scenario. We are not returning anything when exception is not caught.

Reply:
yes, this method is for negative case. If non dictionary is passed, it will throw an error .  If it does not throw an error, test fails if no error is passed